activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -672,7 +672,6 @@ def remove_connection_from_thread_cache(conn, owner_thread = conn.owner)
         def new_connection
           connection = Base.public_send(db_config.adapter_method, db_config.configuration_hash)
           connection.pool = self
```

Comment:
We might be able to get rid of `connection.pool = self` here too, since we already do it later in `adopt_connection`. If I recall correctly it was added so we'd set up the schema cache before calling `check_version`.